src/dataAcces/userRepository.ts

```diff
@@ -0,0 +1,73 @@
+import { PrismaClient, type User } from '@prisma/client';
+import { DbError } from '../errors/errors';
+import { type UserBodyType } from '../models/userModel';
+
+const prisma = new PrismaClient();
+
+export const userRepository = {
+  async getAll() {
+    const dbResult = await prisma.user.findMany();
+
+    return dbResult;
+  },
+
+  async getById(id: number) {
+    const user = await prisma.user.findUnique({
+      where: { id },
+    });
+
+    if (!user) {
+      throw new DbError(`User with ID: ${id} was not found`, 404);
+    }
+
+    return user;
+  },
+
+  async getByLoginOrEmail(login: string, email: string) {
+    const users = await prisma.user.findMany({
+      where: {
+        OR: [{ login }, { email }],
+      },
+    });
+
+    if (users.length > 0) {
+      throw new DbError(`User with same login or email is already exists`, 409);
+    }
+
+    return users;
+  },
+
+  async postUser(user: UserBodyType) {
+    const { login, email, password } = user;
+
+    const newUser = await prisma.user.create({
+      data: { login, email, password },
+    });
+
+    return newUser;
+  },
+
+  async patchUser(user: User) {
+    const { id, login, email, password } = user;
+
+    const updatedUser = await prisma.user.update({
```

Comment:
Please add a check if a user exists and if not then throw an error

Reply:
I have this check in userService.updateUser. Should I move it to userRepository.updateUser?

Reply:
Nah, it's ok it keep in userService